src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -437,10 +437,23 @@ private static bool TryUnfoldITestDataSource(ITestDataSource dataSource, TestDat
         var discoveredTests = new List<UnitTestElement>();
         int index = 0;
 
-        foreach (object?[] d in data)
+        foreach (object?[] dataOrTestDataRow in data)
         {
+            object?[] d = dataOrTestDataRow;
```

Comment:
Do we need `d`? Can't you replace `dataOrTestDataRow`?

Reply:
@Evangelink Assigning to foreach variable produces [CS1656](https://learn.microsoft.com/dotnet/csharp/language-reference/compiler-messages/cs1656) error.